### ISSUE TITLE
Incorrect parameter name in multimapping one-to-one relation

### ISSUE THREAD

**Issue by migueaguilera** [henkmollema/Dommel#104]
Hello, we are running the following code:

```cs
var usuario = await cnx.GetAsync<Usuario, Delegacion, Usuario>(id, (u, d) =>
{
    u.Delegacion = d;
    return u;
});
```
And the resulting SQL query is the following:

```
N'select * from Usuarios left join Delegaciones on Usuarios.IDDelegaciones = Delegaciones.ID
where Usuarios.ID_Usuario = @ID_Usuario',N'@Id int',@Id=93
```

As you can see, the query's parameter name `@ID_Usuario` doesn't match with the supplied parameter `@Id`, causing an error.

The primary key of Usuarios table is `ID_Usuario`.

Best regards.

**Comment by henkmollema** [comment]
Can you share the models of `Usuario` and `Delegacion`?

**Comment by migueaguilera** [comment]
The model classes:

`public class Usuario
{
	public int Id { get; set; }
	public string Login { get; set; }
	public string Password { get; set; }
	public string Nombre { get; set; }
	public string Apellidos { get; set; }    
	public int? DelegacionId { get; set; }
	public Delegacion Delegacion { get; set; }
}

public class Delegacion
{
	public int Id { get; set; }
	public string Codigo { get; set; }
	public string Descripcion { get; set; }
}`

And the mapping:

`public class UsuarioMap : DommelEntityMap<Usuario>
{
	public UsuarioMap()
	{
		ToTable("Usuarios");
		Map(x => x.Id).ToColumn("ID_Usuario");
		Map(x => x.DelegacionId).ToColumn("IDDelegaciones");
	}
}

public class DelegacionMap : DommelEntityMap<Delegacion>
{
	public DelegacionMap()
	{
		ToTable("Delegaciones");
	}
}`

**Comment by henkmollema** [comment]
@migueaguilera I'm wondering where the generated query originates from:

```sql
N'select * from Usuarios left join Delegaciones on Usuarios.IDDelegaciones = Delegaciones.ID
where Usuarios.ID_Usuario = @ID_Usuario',N'@Id int',@Id=93
```

Dommel would (AFAIC) always use `@Id` as parameter name:
https://github.com/henkmollema/Dommel/blob/03534ad65b3d8e094cbd345b429eb45d929ed8a7/src/Dommel/DommelMapper.MultiMap.cs#L765-L768

Also, the parameter order of `GetAsync` is incorrect. The first parameter should be the mapping function and the second is the ID. In your example they seem to be swapped.

**Comment by Wychell** [comment]
I have the same problem


**Model**

```
Entity : Id

 public class Produto : Entity
    {

        public string Nome { get; set; }
        public decimal Valor { get; set; }
        public long CategoriaId { get; set; }
        public Categoria Categoria { get; set; }
}

 public class Categoria : Entity
    {
        public string Nome { get; set; }
    }
```

**Mappings**
```
 public ProdutoMap()
        {
            ToTable("Produto");
            Map(x => x.Nome);
            Map(x => x.CategoriaId);
        }
 public MappingBase()
       {
            Map(x => x.Id).ToColumn(typeof(T).Name + "Id").IsIdentity().IsKey();
       }
```
**Repo** 
```
 using (var db = new SqlConnection(_connectionString))
           {
                return db.Get(id, Mapping());

           }

private static Func<Produto, Categoria, Produto> Mapping()
        {
            return (produto, categoria) =>
            {
                if (categoria != null)
                {
                    produto.Categoria = categoria;
                    produto.CategoriaId = categoria.Id;
                }


                return produto;
            };
        }
```

**Comment by henkmollema** [comment]
Does this issue reproduce on Dommel v2?

**Comment by Wychell** [comment]
Dapper.FluentMap.Dommel Version=1.7.0.0
Dommel Version=1.11.0

**Comment by henkmollema** [comment]
This has been fixed in Dommel 2.0. Unfortunately there is no version of Dapper.FluentMap.Dommel that supports Dommel 2.0 yet. However, support for attribute mapping in is expanded in Dommel 2.0 and you should be able to perform all the column mapping using `[Column]` attributes.
